mesmerize/viewer/modules/roi_manager_modules/managers.py

```diff
@@ -184,22 +183,27 @@ def import_from_imagej(self, path: str):
                 height=ij_roi[k]['height']
                 left=ij_roi[k]['left']
                 top=ij_roi[k]['top']
-                x_center=left+width/2
-                y_center=top+height/2
                 if ij_roi[k]['type'] == 'oval':
-                    shape=Ellipse((x_center,y_center),width,height)
-                    vertices_orig=shape.get_path().vertices #get all vertices  
-                    vertices = shape.get_patch_transform().transform(vertices_orig)
-                    ps=[tuple(x) for x in vertices]
+                    x_bottom=left
+                    y_bottom=top+height
+                    ps=[x_bottom,y_bottom,width,height]
+                    roi= ManualROI.from_ellipse(positions=ps,
+                                           curve_plot_item=self.get_plot_item(),
+                                           view_box=self.vi.viewer.getView())
                 else:
                     ps=list(itertools.product((left,left+width),(top,top+height)))
-                    ps[2],ps[3]=ps[3],ps[2] #swap coordinates so it draws coordinates in sequence
+                    ps[2],ps[3]=ps[3],ps[2] #swap coordinates so it draws coordinates in sequence 
+            elif (ij_roi[k]['type']=='freehand'):
+                #freehand ROIs have large number of datapoints, so reducing it by a fifth
+                xs = ij_roi[k]['x'][::5]
```

Comment:
The freehand ROIs have too many points. Not sure if there is a better way to reduce it?? downsampling??